### PR TITLE
Dis 2

### PR DESCRIPTION
**Pull request by YoniKiriaty**

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/commands/shooter/flyWheel/RunFlyWheelByVelocityWithoutIsFinished.java [hunk 1]
```diff
@@ -2,47 +2,57 @@
...
-		flyWheel.setPreparedToShoot(timeInShootingSpeed >= FlyWheelConstants.MINIMUM_SHOOTING_SPEED_TIME_ROBORIO_CYCLES);
-	}
-
-	@Override
-	public void end(boolean interrupted) {
-		flyWheel.stop();
-	}
+import edu.greenblitz.robotName.utils.hid.SmartJoystick;
+
+public class RunFlyWheelByVelocityWithoutIsFinished extends FlyWheelCommand {
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/shooter/flyWheel/RunFlyWheelByVelocityWithoutIsFinished.java:7]:
```suggestion
public class RunFlyWheelByVelocityUntilInterrupted extends FlyWheelCommand {
```

File: src/main/java/edu/greenblitz/robotName/subsystems/arm/elbow/Elbow.java [hunk 2]
```diff
@@ -16,12 +18,12 @@ public class Elbow extends GBSubsystem {
...
 
     private Rotation2d currentAngle;
-
+    
     private Elbow() {
         elbow = ElbowFactory.create();
         elbowInputs = new ElbowInputsAutoLogged();
         elbow.updateInputs(elbowInputs);
-        currentAngle = elbowInputs.position;
+        currentAngle = NeoElbowConstants.MINIMUM_ANGLE;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/arm/elbow/Elbow.java:26]: why?

File: src/main/java/edu/greenblitz/robotName/subsystems/arm/elbow/Elbow.java [hunk 3]
```diff
@@ -97,6 +102,11 @@ public boolean isInShooterCollisionRange() {
                 elbowInputs.position.getRadians() <= ElbowConstants.SHOOTER_COLLISION_RANGE.getSecond().getRadians();
     }
 
+    public void moveToSafeAngleInit(){
+        if(isInShooterCollisionRange()){
+            moveToAngle(ElbowConstants.PresetPositions.SAFE.ANGLE);
+        }
+    }
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/arm/elbow/Elbow.java:109]: add a space

File: src/main/java/edu/greenblitz/robotName/subsystems/arm/elbow/neoElbow/NeoElbow.java [hunk 4]
```diff
@@ -53,13 +47,16 @@ public void setIdleMode(NeutralModeValue idleMode) {
 
     @Override
     public void resetAngle(Rotation2d position) {
-        motor.getEncoder().setPosition(position.getRadians());
+        motor.getEncoder().setPosition(position.getRotations());
     }
 
     @Override
     public void moveToAngle(Rotation2d targetAngle) {
+        if (targetAngle.getDegrees() <= ElbowConstants.BACKWARD_ANGLE_LIMIT.getDegrees()){
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/arm/elbow/neoElbow/NeoElbow.java:55]: what aout forward limit?

File: src/main/java/edu/greenblitz/robotName/subsystems/arm/elbow/neoElbow/NeoElbowConstants.java [hunk 5]
```diff
@@ -1,39 +1,42 @@
...
-
     public static final int PID_SLOT = 0;
 
-    public static final PIDObject PID = new PIDObject().withKp(0.8).withKd(0.3).withKi(0).withMaxPower(1);
+    public static final PIDObject PID = new PIDObject().withKp(2).withKd(0).withKi(0).withMaxPower(0.5);
 
     public static final GBSparkMax.SparkMaxConfObject ELBOW_CONFIG_OBJECT = new GBSparkMax.SparkMaxConfObject()
             .withPID(PID)
-            .withIdleMode(CANSparkMax.IdleMode.kBrake)
+            .withIdleMode(CANSparkMax.IdleMode.kCoast)
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/arm/elbow/neoElbow/NeoElbowConstants.java:36]: why?

File: src/main/java/edu/greenblitz/robotName/subsystems/lifter/LifterConstants.java [hunk 6]
```diff
@@ -16,7 +16,7 @@ public class LifterConstants {
 
     public static final double POWER_TO_OPEN_SOLENOID = 0;
 
-    public static final double POWER_TO_CLOSE_SOLENOID = 0.95;
+    public static final double POWER_TO_CLOSE_SOLENOID = 1;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/lifter/LifterConstants.java:19]: are we sure about this?

File: src/main/java/edu/greenblitz/robotName/subsystems/lifter/neoLifter/NeoLifterConstants.java [hunk 7]
```diff
@@ -25,7 +24,6 @@ public class NeoLifterConstants {
     public static final SparkLimitSwitch.Type BACKWARD_LIMIT_SWITCH_TYPE = SparkLimitSwitch.Type.kNormallyClosed;
 
     public static final boolean IS_BACKWARD_LIMIT_SWITCH_ENABLED = true;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/lifter/neoLifter/NeoLifterConstants.java:26]: give the space back

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FlyWheelInputs.java [hunk 8]
```diff
@@ -20,4 +20,6 @@ public class FlyWheelInputs {
 	public double leftFlywheelCurrent;
 	
 	public double leftWheelAcceleration;
+	public double rightVelocityReferance;
+	public double leftVelocityReferance;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FlyWheelInputs.java:24]: sapces

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 9]
```diff
@@ -26,111 +22,111 @@
...
 	public void initButtons() {
 		romyButtons();
 		shchoriButtons();
 	}
-
+	
 	public void romyButtons() {
-		mainJoystick.R1.whileTrue(new CollectNoteFromGround());
-		mainJoystick.L1.whileTrue(new MoveRobotToShootingPosition(ShootingPositionConstants.OPTIMAL_SHOOTING_ZONE));
+		mainJoystick.R1.whileTrue(new NoteToShooter());
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/OI.java:79]: isnt this also from gorund? consider re-naming

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/funnel/neoFunnel/NeoFunnelConstants.java [hunk 10]
```diff
@@ -44,5 +44,5 @@ public class NeoFunnelConstants {
 			.withCurrentLimit(CURRENT_LIMIT)
 			.withInverted(IS_INVERTED)
 			.withPID(PID_CONTROLLER)
-			.withVelocityConversionFactor(1 / 5.0);
+			.withVelocityConversionFactor(1 /9.0);
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/funnel/neoFunnel/NeoFunnelConstants.java:47]: give the space back

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivotConstants.java [hunk 11]
```diff
@@ -80,9 +80,18 @@ public class FalconPivotConstants {
 		HARDWARE_LIMIT_SWITCH_CONFIGS.ForwardLimitEnable = false;
 	}
 	
+	public static final  ClosedLoopRampsConfigs CLOSED_LOOP_RAMPS_CONFIGS = new ClosedLoopRampsConfigs();
+	
+	static {
+		CLOSED_LOOP_RAMPS_CONFIGS.VoltageClosedLoopRampPeriod = 0.5;
+	}
+	
+
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivotConstants.java:89]: delete

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/chassis/SwerveChassis.java [hunk 12]
```diff
@@ -130,8 +130,9 @@ public void periodic() {
 		Logger.processInputs("DriveTrain/Chassis", ChassisInputs);
 		Logger.processInputs("DriveTrain/Gyro", gyroInputs);
 
-//		updatePoseEstimationLimeLight();
 		updatePoseEstimatorOdometry();
+//		updatePoseEstimationLimeLight();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/chassis/SwerveChassis.java:134]: ...

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/chassis/SwerveChassis.java [hunk 13]
```diff
@@ -86,7 +86,7 @@ public SwerveChassis() {
 				this.kinematics,
 				getGyroAngle(),
 				getSwerveModulePositions(),
-				visionPoseStartMatch(),
+				new Pose2d(),
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/chassis/SwerveChassis.java:89]: i am pretty sure we fixed that already, didnt we?

File: src/main/java/edu/greenblitz/robotName/utils/AutonomousSelector.java [hunk 14]
```diff
@@ -17,6 +18,7 @@ public class AutonomousSelector {
 
     private AutonomousSelector() {
         chooser = AutoBuilder.buildAutoChooser();
+        chooser.addOption("playoffsgame", new ShootToSpeakerFromClose());
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/utils/AutonomousSelector.java:21]: no

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/Robot.java [hunk 15]
```diff
@@ -98,25 +86,28 @@ public void initializeSubsystems() {
...
-
     }
 
     private void initializeAutonomousBuilder() {
-        NamedCommands.registerCommand("shoot", new ShootFromInFunnel());
-        NamedCommands.registerCommand("close shoot", new ShootToSpeakerFromClose());
-        NamedCommands.registerCommand("grip",(new CollectNoteFromGround()));
+//        NamedCommands.registerCommand("shoot", new ShootFromInFunnel());
+//        NamedCommands.registerCommand("close shoot", new ShootToSpeakerFromClose());
+//        NamedCommands.registerCommand("grip", new NoteToShooter());
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/Robot.java:104]: ...

File: src/main/java/edu/greenblitz/robotName/commands/arm/elbow/SafeMoveElbowToAngle.java [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+
+    @Override
+    public void initialize() {
+        isElbowCrashableIntoWall = SwerveChassis.getInstance().isRobotNearBoundsOfField();
+        if (isElbowCrashableIntoWall) {
+            elbow.setCurrentAngle();
+            elbow.standInPlace();
+        }
+        else {
+        elbow.moveToAngle(targetAngle);
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/arm/elbow/SafeMoveElbowToAngle.java:30]: add a tab

File: src/main/java/edu/greenblitz/robotName/utils/AutonomousSelector.java [hunk 17]
```diff
@@ -18,6 +19,7 @@ public class AutonomousSelector {
     private AutonomousSelector() {
         chooser = AutoBuilder.buildAutoChooser();
         ShuffleboardTab tab = Shuffleboard.getTab("auto");
+        chooser.addOption("auto", AutoBuilder.buildAuto("2 to 1 close"));
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/utils/AutonomousSelector.java:22]: remove

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/commands/arm/elbow/SafeMoveElbowToAngle.java [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+        else {
+            elbowMoveToAngleAccordingToRobotType();
+        }
+    }
+
+    private void elbowStandInPlaceAccordingToRobotType() {
+        if (!isElbowCrashableIntoWall || Robot.isSimulation()) {
+            elbow.setCurrentAngle();
+            elbow.standInPlace();
+            isElbowCrashableIntoWall = true;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/commands/arm/elbow/SafeMoveElbowToAngle.java:48]: Relocate this assignment to the condition itself and not in a void function

**Review by yonnbro — APPROVED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Pull request merged**